### PR TITLE
IGAPP-757: Increase connectionRetryTimeout in E2E Tests

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

As always this one cannot guarante, that there will be no more failing tests. But I ran both the ios and the android tests quite often to ensure that they do not time out in 1 of 5 runs, which is about 90% better what we currently have. As I just said increasing the timeout seems to be the most sufficient solution here, as I did try a few other things, too.

### CODE REVIEW

File: e2e-tests/native/capabilities.ts [hunk 1]
```diff
@@ -10,15 +10,16 @@ const browserstackCaps = (
...
   const app = process.env.E2E_BROWSERSTACK_APP
+
   return {
     'bstack:options': {
       buildName: `${prefix}: ${getGitBranch()}`,
       sessionName: `${platformName.toLowerCase()}: ${getGitHeadReference()}`,
       projectName: 'integreat-app-native',
       debug: true,
       realMobile: isCi,
-      networkProfile: '4g-lte-good',
```
Comment (Taggotty) [inline e2e-tests/native/capabilities.ts:20]: was an experiment to get rid of timeouts in Android, which did not work, therefore I removed it.

File: e2e-tests/native/capabilities.ts [hunk 2]
```diff
@@ -41,7 +42,7 @@ export default {
       'appium:deviceName': 'iPhone 8',
       'appium:automationName': 'XCUITest',
       // @ts-ignore unknown property
-      'appium:waitForQuiescence': true
+      'appium:waitForIdleTimeout': 10000
```
Comment (Taggotty) [inline e2e-tests/native/capabilities.ts:45]: `waitForIdleTimeout` is very similar to `waitForQuiescence` as it waits for the timeout or idle by itself. I replaced those two as webdriver seems to translate unknown booleans to strings -.-

File: e2e-tests/native/wdio-browserstack.conf.ts [hunk 3]
```diff
@@ -13,32 +15,31 @@ const getCapability = () => {
...
   user: process.env.E2E_BROWSERSTACK_USER,
   key: process.env.E2E_BROWSERSTACK_KEY,
 
   capabilities: [getCapability()],
 
   logLevel: 'info',
-  coloredLogs: true,
-  bail: 0,
+
+  bail: 1,
```
Comment (Taggotty) [inline e2e-tests/native/wdio-browserstack.conf.ts:32]: If now one test fails in a suite, the other tests wont be executed,

File: e2e-tests/native/wdio-browserstack.conf.ts [hunk 4]
```diff
@@ -13,32 +15,31 @@ const getCapability = () => {
...
   baseUrl: 'http://localhost:9000',
   waitforTimeout: 100000,
   waitforInterval: 2000,
-  connectionRetryTimeout: 50000,
-  connectionRetryCount: 3,
-  services: [['browserstack']],
-  host: 'hub.browserstack.com',
+  connectionRetryTimeout: 120000,
+  connectionRetryCount: 2,
```
Comment (Taggotty) [inline e2e-tests/native/wdio-browserstack.conf.ts:37]: After a retry, the old session is not deleted or terminated immidiately, which can lead to a hand full of ghost sessions, when we retry to often. Decreasing the idle timeout (see Capabilities) and the connectionRetryCount helps cope with this.

**Review by SteffiStuffel — APPROVED** [review]

**Review by klinzo — APPROVED** [review]

**Pull request merged**